azurerm/resource_arm_storage_account.go

```diff
@@ -825,6 +865,14 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
 		return err
 	}
 
+	advancedThreatProtectionSetting, err := advancedThreatProtectionClient.Get(ctx, d.Id())
+
+	if err != nil {
+		return fmt.Errorf("Error reading the advanced threat protection settings of AzureRM Storage Account %q: %+v", name, err)
+	}
+
+	d.Set("enable_advanced_threat_protection", advancedThreatProtectionSetting.AdvancedThreatProtectionProperties.IsEnabled)
```

Comment:
We will need to check for nils here:
```suggestion
	if advancedThreatProtectionSetting != nil {
		if atpp := advancedThreatProtectionSetting.AdvancedThreatProtectionProperties; atpp != nil {
			d.Set("enable_advanced_threat_protection", atpp.IsEnabled)
		}
	} 
```

Reply:
done. Just one question about the first nil check. In all the other client calls the results are not checked for nil. E. g. `resp, err := client.GetProperties(ctx, resGroup, name, "")` in line `764`. Can you tell me why this check is required and the other one not?

Reply:
Remove the first nil check again. `advancedThreatProtectionSetting` is not a pointer, so `&advancedThreatProtectionSetting` can't be nil. This also explains my question in the previous comment.

Reply:
My bad, advancedThreatProtectionSetting not being resp threw me off, this is good now 🙂